### PR TITLE
Add refresh tokens

### PR DESCRIPTION
**Pull request by calebeby**
You can test this by verifying that `refreshToken` is created in localstorage, and that if you delete the `jwt`, it is automatically re-populated by making a request to `/refresh`

https://refresh-tokens--peregrine.netlify.com/

### CODE REVIEW

File: src/jwt.ts [hunk 1]
```diff
@@ -10,19 +11,42 @@ interface JWT {
   raw: string
 }
 
+const isExpired = (jwt: JWT) => new Date(jwt.exp * 1000) < new Date()
```
Comment (fharding1) [inline src/jwt.ts]: I think this should be <= and you should compare by casting `new Date()` to a number.

```
  The "exp" (expiration time) claim identifies the expiration time on
   or after which the JWT MUST NOT be accepted for processing.  The
   processing of the "exp" claim requires that the current date/time
   MUST be before the expiration date/time listed in the "exp" claim.```

Comment (calebeby) [reply src/jwt.ts]: You can compare dates without casting to numbers

Comment (fharding1) [reply src/jwt.ts]: I know, but if the expiry time is specified as a unix time it only specifies the precision to a second, and a date has more precision than that.

Comment (calebeby) [reply src/jwt.ts]: Won't it just expire up to a second early on the FE?

Comment (fharding1) [reply src/jwt.ts]: Yeah

Comment (calebeby) [reply src/jwt.ts]: Is that a problem?

Comment (fharding1) [reply src/jwt.ts]: I mean it's an off-by-one. It's not a huge problem, but it's technically wrong and it's easy to fix, so IMO you might as well fix it.

**Pull request merged**